Update cds-typer doc (#1263)
cap-js/docs: tools/cds-typer.md

    If specified, cds-typer will create a jsconfig.json file and
    set it up to restrict property usage in types entities to
    existing properties only.

  --logLevel
  --log_level SILENT | ERROR | WARN | INFO | DEBUG | TRACE | SILLY | VERBOSE
    (default: ERROR)

    Minimum log level that is printed.
    The default is only used if no explicit value is passed
    and there is no configuration passed via cds.env either.

  --outputDirectory
  --output_directory: <string>
    (default: ./)

    Root directory to write the generated files to.

  --propertiesOptional
  --properties_optional: <true | false>
    (default: true)

    If set to true, properties in entities are
    always generated as optional (a?: T).

  --useEntitiesProxy
  --use_entities_proxy: <true | false>
    (default: false)

    If set to true the 'cds.entities' exports in the generated 'index.js'
    files will be wrapped in 'Proxy' objects
    so static import/require calls can be used everywhere.
    
    WARNING: entity properties can still only be accessed after
    'cds.entities' has been loaded

  --version

    Prints the version of this tool.
```
:::

### Configuration

> Any CLI parameter described [above](#typer-cli) can also be passed to cds-typer via [`cds.env`](#cds-env), for example via your project's _package.json_:

Reviewer: ```suggestion
Any CLI parameter described [above](#typer-cli) can also be passed to cds-typer via [`cds.env`](../node.js/cds-env), for example via your project's _package.json_:
```

::: code-group
```json [package.json]
{
  …
  "cds": {
    "typer": {

>       "propertiesOptional": true,

Reviewer: Should promote the snake-case options:

```suggestion
      "properties_optional": true,
```

Author: I wanted to showcase that both forms are still valid. Is there harm to be expected from users using camelCase?

Reviewer: > I wanted to showcase that both forms are still valid. Is there harm to be expected from users using camelCase?

No harm.  Just wondered if it would look inconsistent to see both styles.  In the end, users will ask which style they are supposed to use.

Author: > which style they are supposed to use
Whichever they fancy. 🙂

But if we want to nudge them to snake_case, let's remove the one with camelCase altogether from the sample. No need for two examples then.

Reviewer: > But if we want to nudge them to snake_case,

Yes, kind of, for `cds.env`.  Because we tell in other areas to use this style to be able to use env vars, i.e. `CDS_TYPER_PROPERTIES__OPTIONAL` here.

Author: ACK, I have removed the camelCase example accordingly.

})
```

> Just as with `cds.entities(…)`, these imports can't be static, but need to be dynamic (caveat, [see below](#typer-top-level-imports)):

Reviewer 2: ```suggestion
Similar to `cds.entities(…)`, you can't use static imports here. Instead, you need to use dynamic imports. However, there's an exception for [static top-level imports](#typer-top-level-imports).
```

  const { Book } = await import('#cds-models/sap/capire/bookshop')
}}
```

### Static Top-Level Imports {#typer-top-level-imports}

> Starting with `cds-typer@0.26.0`, a new option `useEntitiesProxy` can be passed to `cds-typer` that allows you to statically import your entities on top-level, as you intuitively would. You can still only _use them_ in a context where the cds runtime is fully booted, like in a service definition:

Reviewer 2: ```suggestion
Starting with `cds-typer@0.26.0`, you can pass a new option, `useEntitiesProxy`, to `cds-typer`. This option allows you to statically import your entities at the top level, as you intuitively would. However, you can still only _use these entities_ in a context where the CDS runtime is fully booted, like in a service definition:
```
